python/unblob/testing.py

```diff
@@ -222,14 +223,44 @@ def start(self, s):
 
 
 def is_sandbox_available():
+    exception = None
     is_sandbox_available = True
 
-    try:
-        restrict_access(AccessFS.read_write("/"))
-    except SandboxError:
-        is_sandbox_available = False
-
-    if platform.architecture == "x86_64" and platform.system == "linux":
-        assert is_sandbox_available, "Sandboxing should work at least on Linux-x86_64"
-
+    def _run_catching_exceptions(fn):
+        def wrapper():
+            __tracebackhide__ = True
+            nonlocal exception
+            try:
+                fn()
+            except BaseException as exc:
+                exception = exc
+
+        return wrapper
+
+    # Sandbox applies to the current thread and future threads spawned
+    # from it.
+    #
+    # Running the test on a new thread keeps the main-thread
+    # clean, so sandboxing won't interfere with other tests executed
+    # after this one.
+
+    @_run_catching_exceptions
+    def _run_in_thread():
+        nonlocal is_sandbox_available
+        try:
+            restrict_access(AccessFS.read("/"))
+        except SandboxError:
+            is_sandbox_available = False
+
+        if platform.architecture == "x86_64" and platform.system == "linux":
+            assert is_sandbox_available, (
+                "Sandboxing should work at least on Linux-x86_64"
+            )
+
+    t = threading.Thread(target=_run_in_thread)
+    t.start()
+    t.join()
+    __tracebackhide__ = True
+    if exception:
+        raise exception
     return is_sandbox_available
```

Comment:
I think this mode of checking would be cleaner:
https://github.com/onekey-sec/unblob/blob/main/tests/rust/test_sandbox.py#L38-L69

Reply:
See https://github.com/onekey-sec/unblob/pull/1129/commits/ced1d96e505ecaa89efcbf141e457d6d15ca1c70